### PR TITLE
JBDS-4410 Add download size for packages

### PR DESCRIPTION
**Pull request by mohitsuman**
https://issues.jboss.org/browse/JBDS-4410

Added the download size information with the version name. Mockup is attached to the jira.

### CODE REVIEW

**Comment by dgolovin** [comment]
test this

**Comment by dgolovin** [comment]
test this

**Comment by dgolovin** [comment]
test this

**Comment by dgolovin** [comment]
test this

File: requirements.json [hunk 1]
```diff
@@ -15,7 +15,8 @@
     "sha256sum": "59560c89b76f57ee72d9bd20bac047146f6f369fb42fba94ebaccd65c71209f5",
     "version": "11.0.0.GA",
     "useDownload": true,
-    "requires": ["jdk"]
+    "requires": ["jdk"],
+    "size": "645.38 MB"
```
Comment (dgolovin) [inline requirements.json]: We should agree on units we use in size field (lets say Kilobytes) and put only number into this field. That would simplify total download computation. On UI side I would use angular filter and transform the number with humanize module to readable value.

Filter would look like:

``` JavaScript
module.filter('humanReadableSize', function() {
    return function(input) {
        return humanize.filesize(input*1000); //be careful with with kilobyte (x1000) and Kibibyte (x1024).
    }
});
```

then on template side:

``` html
{{ item.size | humanReadableSize }}
```

File: browser/model/installable-item.js [hunk 2]
```diff
@@ -72,6 +72,10 @@ class InstallableItem {
     return this.productVersion;
   }
 
+  getProductSize() {
```
Comment (mohitsuman) [inline browser/model/installable-item.js]: I have added the function as to calculate total download size which we will show. Do you want to get this code in that PR. I will make that change

Comment (dgolovin) [inline browser/model/installable-item.js]: It is up to you, but if you add new function, please, add unit test for that as well.

Comment (mohitsuman) [inline browser/model/installable-item.js]: Removed the function as of now. Will use this for calculating total download size.

**Comment by dgolovin** [comment]
test this

**Comment by mohitsuman** [comment]
test this

**Comment by dgolovin** [comment]
test this

File: browser/directives/componentPanel.html [hunk 3]
```diff
@@ -25,7 +25,8 @@
             </div>
           </div>
         </div>
-        <span id="{{item.keyName}}-name" class="product-name">{{item.productName}}</span><span id="{{item.keyName}}-version" class="product-version">{{item.version}}</span>
+        <span id="{{item.keyName}}-name" class="product-name">{{item.productName}}</span>
+        <span id="{{item.keyName}}-version" class="product-info">{{item.version}}&nbsp;|&nbsp;{{item.size}}</span>
```
Comment (dgolovin) [inline browser/directives/componentPanel.html]: UI test errors related to change n this line. adding span tag per text element w/ or w/o id attribute 

``` html
<span id="{{item.keyName}}-version" class="product-info">{{item.version}}</span><span class="product-info">|</span><span id="{{item.keyName}}-download-size" class="product-info">{{item.size}}</span>
```

fixes the UI test errors.

**Comment by jrichter1** [comment]
Technical question: how do you intend to keep these values fresh? (Because I know someone will forget eventually)
Maybe a build time script? Plus we could use this value instead of calculating it again when downloads start. Doesn't need to be a part of this PR though.

**Comment by mohitsuman** [comment]
test this

**Comment by dgolovin** [comment]
@jrichter1 the same way we update devstudio info with latest nightly build available. That is different issue to address later.

**Pull request merged**